bluepysnap/edges/edge_population.py

```diff
@@ -60,12 +60,20 @@ def __init__(self, circuit, population_name):
         self._circuit = circuit
         self.name = population_name
 
-    @property
+    @cached_property
     def _properties(self):
+        """Edge population properties.
+
+        Cached for some performance improvement when it's accessed several times.
+        """
         return self._circuit.to_libsonata.edge_population_properties(self.name)
 
-    @property
+    @cached_property
     def _population(self):
+        """Libsonata edge population.
+
+        Cached for some performance improvement when it's accessed several times.
+        """
```

Comment:
won't this potentially keep a handle to the hdf5 file open? I'm a bit afraid of that, because other hdf5 consumers may exist in the same process, and then errors may be emitted

could you check?

Reply:
You're right, the libsonata population is keeping the h5 file open for reading. I verified with:
```python
import psutil
import libsonata
path = "/gpfs/bbp.cscs.ch/project/proj12/SIT/thalamus_sonata/circuit_sonata.json"
pop = libsonata.CircuitConfig.from_file(path).node_population("thalamus_neurons")
p = psutil.Process()
for i in p.open_files():
    print(i)
```
and I can see, unless pop is deleted:
```
popenfile(path='/gpfs/bbp.cscs.ch/project/proj12/SIT/thalamus_sonata/networks/nodes/thalamus_neurons/nodes.h5', fd=11, position=0, mode='r', flags=32768)
```
However, do you know how it's possible to reproduce errors when multiple hdf5 readers exist in the same process?

Reply:
> However, do you know how it's possible to reproduce errors when multiple hdf5 readers exist in the same process?

Unfortunately no; in theory, I think multiple readers are safe, but I don't know if that depends on how hdf5 is compiled, or whether there are any other gotchas.

Reply:
ic, I reverted the changes and added a small explanation in the docstring.
I also used self. _properties to simplify a bit the code in spatial_segment_index and spatial_synapse_index